controllers/taloscontrolplane_controller.go

```diff
@@ -205,46 +213,37 @@ func (r *TalosControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 	controlPlane := newControlPlane(cluster, tcp, ownedMachines)
 
-	// If we've made it this far, we can assume that all ownedMachines are up to date
-	numMachines := len(ownedMachines)
-	desiredReplicas := int(*tcp.Spec.Replicas)
-
-	requeue := false
-
-	// Audit the etcd member list to remove any nodes that no longer exist
-	if err := r.auditEtcd(ctx, util.ObjectKey(cluster), controlPlane.TCP.Name); err != nil {
-		logger.Info("failed to check etcd membership list", "error", err)
-
-		// if audit failed, requeue the reconcile in any case
-		requeue = true
-	}
+	conditionGetters := make([]conditions.Getter, len(ownedMachines))
 
-	if err := r.etcdHealthcheck(ctx, cluster, ownedMachines); err != nil {
-		conditions.MarkFalse(tcp, controlplanev1.EtcdClusterHealthyCondition, controlplanev1.EtcdClusterUnhealthyReason,
-			clusterv1.ConditionSeverityWarning, err.Error())
-	} else {
-		conditions.MarkTrue(tcp, controlplanev1.EtcdClusterHealthyCondition)
+	for i, v := range ownedMachines {
+		conditionGetters[i] = &v
 	}
 
-	if err := r.nodesHealthcheck(ctx, cluster, ownedMachines); err != nil {
-		reason := controlplanev1.ControlPlaneComponentsInspectionFailedReason
+	conditions.SetAggregate(tcp, controlplanev1.MachinesReadyCondition, conditionGetters, conditions.AddSourceRef(), conditions.WithStepCounterIf(false))
 
-		if errors.Is(err, &errServiceUnhealthy{}) {
-			reason = controlplanev1.ControlPlaneComponentsUnhealthyReason
-		}
+	// If we've made it this far, we can assume that all ownedMachines are up to date
+	numMachines := len(ownedMachines)
+	desiredReplicas := int(*tcp.Spec.Replicas)
 
-		conditions.MarkFalse(tcp, controlplanev1.ControlPlaneComponentsHealthyCondition, reason,
-			clusterv1.ConditionSeverityWarning, err.Error())
-	} else {
-		conditions.MarkTrue(tcp, controlplanev1.ControlPlaneComponentsHealthyCondition)
-	}
+	var (
+		errs        error
+		result      ctrl.Result
+		phaseResult ctrl.Result
+	)
 
-	if !conditions.Has(tcp, controlplanev1.AvailableCondition) {
-		conditions.MarkFalse(tcp, controlplanev1.AvailableCondition, controlplanev1.WaitingForTalosBootReason, clusterv1.ConditionSeverityInfo, "")
-	}
+	// run all similar reconcile steps in the loop and pick the lowest RetryAfter, aggregate errors and check the requeue flags.
+	for _, phase := range []func(context.Context, *clusterv1.Cluster, *controlplanev1.TalosControlPlane, []clusterv1.Machine) (ctrl.Result, error){
+		r.reconcileEtcdMembers,
+		r.reconcileNodeHealth,
+		r.reconcileConditions,
+		r.reconcileKubeconfig,
+	} {
+		phaseResult, err = phase(ctx, cluster, tcp, ownedMachines)
+		if err != nil {
+			errs = kerrors.NewAggregate([]error{errs, err})
+		}
 
-	if !conditions.Has(tcp, controlplanev1.MachinesBootstrapped) {
-		conditions.MarkFalse(tcp, controlplanev1.MachinesBootstrapped, controlplanev1.WaitingForMachinesReason, clusterv1.ConditionSeverityInfo, "")
+		result = util.LowestNonZeroResult(result, phaseResult)
```

Comment:
if we set result here, and later move to the next "scale" block, are we losing the `result` ?

Reply:
yep, fixed that